presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java

```diff
@@ -1187,6 +1189,18 @@ public SystemSessionProperties(
                         KEY_BASED_SAMPLING_FUNCTION,
                         "Sampling function for key based sampling",
                         "key_sampling_percent",
+                        false),
+                integerProperty(
+                        HASH_BASED_DISTINCT_LIMIT_THRESHOLD,
+                        "threshold for doing hash based distinct",
+                        // 10k is a decent-sized limit that guarantees almost no collisions if the # distinct < 10k according to:
+                        //   https://stackoverflow.com/questions/22029012/probability-of-64bit-hash-code-collisions
+                        10000,
+                        false),
+                booleanProperty(
+                        HASH_BASED_DISTINCT_LIMIT_ENABLED,
+                        "Hash based distinct limit enabled",
```

Comment:
Would users understand what "Hash based distinct limit" means and what is "threshold" for that? Should we add some documentation somewhere in https://prestodb.io/docs/current/optimizer.html and/or https://prestodb.io/docs/current/admin/properties.html#optimizer-properties ?

Reply:
It's intentional. We are not expecting users to mess with this. I'm thinking this will be mostly enable/disable by admins for whole clusters - enable it for adhoc/analytical clusters only.

Reply:
Is there reasons why we don't want to enable this for batch workloads? My thinking is that we are adding the boolean config to control roll out, but once this is tested, we can actually remove it, and always use the threshold (assuming that under certain numbers, it would always be beneficial, say 100)

Reply:
Sure we can. It's upto whoever wants to use it :) Also generally we don't see this pattern in batch workloads.